commit 45f332d08c79204030e4f7608e166d0eb10a3a92
Author: Alessandro Gario <alessandro.gario@gmail.com>
Date:   Fri Oct 11 23:15:45 2019 +0200

    CMake: Add a basic project

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..9348c88
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.10)
+project(openssl)
+set(ENV{CC} ${CMAKE_C_COMPILER})
+set(ENV{LD} ${CMAKE_LINKER})
+execute_process(
+    COMMAND ./config no-shared enable-static-engine no-ssl2 no-idea no-bf no-cast no-seed no-md2 "--prefix=${CMAKE_INSTALL_PREFIX}" --openssldir=/etc/ssl
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}")
+
+execute_process(
+  COMMAND /usr/bin/sed -i ".bak" "s+^CFLAGS=+CFLAGS=-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk +g" "${CMAKE_SOURCE_DIR}/Makefile"
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}")
+
+add_custom_command(
+    OUTPUT make_all.stamp
+    COMMAND ${CMAKE_MAKE_PROGRAM} all
+    COMMAND "${CMAKE_COMMAND}" -E touch make_all.stamp
+    COMMENT "Building OpenSSL"
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}")
+
+add_custom_target(make_all ALL DEPENDS make_all.stamp)
+
+install(CODE "execute_process(COMMAND make install_sw WORKING_DIRECTORY \"${CMAKE_SOURCE_DIR}\")")
+
